server/src-lib/Hasura/Eventing/HTTP.hs

```diff
@@ -0,0 +1,363 @@
+  toEngineLog req = (LevelInfo, eventTriggerLogType, J.toJSON req)
+
+logHTTPForET
+  :: ( MonadReader r m
+     , Has (Logger Hasura) r
+     , MonadIO m
+     )
+  => Either (HTTPErr 'ET) (HTTPResp 'ET) -> Maybe ExtraLogContext -> m ()
+logHTTPForET eitherResp extraLogCtx = do
+  logger :: Logger Hasura <- asks getter
+  unLogger logger $ HTTPRespExtra eitherResp extraLogCtx
+
+logHTTPForST
+  :: ( MonadReader r m
+     , Has (Logger Hasura) r
+     , MonadIO m
+     )
+  => Either (HTTPErr 'ST) (HTTPResp 'ST) -> Maybe ExtraLogContext -> m ()
+logHTTPForST eitherResp extraLogCtx = do
+  logger :: Logger Hasura <- asks getter
+  unLogger logger $ HTTPRespExtra eitherResp extraLogCtx
+
+runHTTP :: (MonadIO m) => HTTP.Manager -> HTTP.Request -> m (Either (HTTPErr a) (HTTPResp a))
+runHTTP manager req = do
+  res <- liftIO $ try $ HTTP.httpLbs req manager
+  return $ either (Left . HClient) anyBodyParser res
+
+tryWebhook ::
+  ( MonadReader r m
+  , Has HTTP.Manager r
+  , MonadIO m
+  , MonadError (HTTPErr a) m
+  )
+  => [HTTP.Header]
+  -> HTTP.ResponseTimeout
+  -> J.Value
+  -> String
+  -> m (HTTPResp a)
+tryWebhook headers timeout payload webhook = do
+  initReqE <- liftIO $ try $ HTTP.parseRequest webhook
+  manager <- asks getter
+  case initReqE of
+    Left excp -> throwError $ HClient excp
+    Right initReq -> do
+      let req =
+            initReq
+              { HTTP.method = "POST"
+              , HTTP.requestHeaders = headers
+              , HTTP.requestBody = HTTP.RequestBodyLBS (J.encode payload)
+              , HTTP.responseTimeout = timeout
+              }
+      eitherResp <- runHTTP manager req
+      onLeft eitherResp throwError
+
+data TriggerMetadata
+  = TriggerMetadata { tmName :: TriggerName }
+  deriving (Show, Eq)
+
+$(J.deriveJSON (J.aesonDrop 2 J.snakeCase){J.omitNothingFields=True} ''TriggerMetadata)
+
+data DeliveryInfo
+  = DeliveryInfo
+  { diCurrentRetry :: Int
+  , diMaxRetries   :: Int
+  } deriving (Show, Eq)
+
+$(J.deriveJSON (J.aesonDrop 2 J.snakeCase){J.omitNothingFields=True} ''DeliveryInfo)
+
+mkResp :: Int -> TBS.TByteString -> [HeaderConf] -> Response
+mkResp status payload headers =
+  let wr = WebhookResponse payload (mkMaybe headers) status
+  in ResponseType1 wr
+
+mkClientErr :: TBS.TByteString -> Response
+mkClientErr message =
+  let cerr = ClientError message
+  in ResponseType2 cerr
+
+mkWebhookReq :: J.Value -> [HeaderConf] -> Version -> WebhookRequest
+mkWebhookReq payload headers = WebhookRequest payload (mkMaybe headers)
+
+isClientError :: Int -> Bool
+isClientError status = status >= 1000
+
+mkMaybe :: [a] -> Maybe [a]
+mkMaybe [] = Nothing
+mkMaybe x  = Just x
+
+-- logHTTPErr
+--   :: ( MonadReader r m
+--      , Has (L.Logger L.Hasura) r
+--      , MonadIO m
+--      )
+--   => HTTPErr a -> m ()
+-- logHTTPErr err = do
+--   logger :: L.Logger L.Hasura <- asks getter
+--   L.unLogger logger err
+
+toInt64 :: (Integral a) => a -> Int64
+toInt64 = fromIntegral
```

Comment:
Because our sql arguments can only be `Int64`, `Int32` or `Int16` : https://github.com/hasura/pg-client-hs/blob/70a849d09bea9461e72c5a5bbde06df65aab61c0/src/Database/PG/Query/Class.hs#L439

Reply:
What is the advantage over writing `fromIntegral @_ @Int64`?

Reply:
This is an outdated diff, can you check the latest code. Did you mean type cast it explicitly at the callsite when you said `fromIntegral @_ @Int64`